geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionTest.java

```diff
@@ -573,6 +578,50 @@ public void transactionThrowsTransactionDataRebalancedExceptionIfIsAForceReattem
         .hasMessage(PartitionedRegion.DATA_MOVED_BY_REBALANCE).hasCause(exception);
   }
 
+  @Test
+  public void failuresSavedIfFetchKeysThrows() throws Exception {
+    PartitionedRegion spyPartitionedRegion = spy(partitionedRegion);
+
+    VersionedObjectList values = mock(VersionedObjectList.class);
+    ServerConnection serverConnection = mock(ServerConnection.class);
+    Set<Integer> failures = mock(Set.class);
```

Comment:
It's best not to mock Collections. It's fine to just use an actual Set here.

Reply:
The mocked failures as a set is used later to verify whether a method is invoked on the set. Otherwise it would be hard to verify.

Reply:
Instead of verifying the method call on the mocked set, you can verify the contents of the real set, which should contain one item, which is the `Integer` 1.